core/server/worker/src/main/java/alluxio/worker/block/SpaceReserver.java

```diff
@@ -59,25 +61,38 @@ public SpaceReserver(BlockWorker blockWorker) {
     Map<String, Long> capOnTiers = blockWorker.getStoreMeta().getCapacityBytesOnTiers();
     long lastTierReservedBytes = 0;
     for (int ordinal = 0; ordinal < mStorageTierAssoc.size(); ordinal++) {
-      PropertyKey tierReservedSpaceProp =
-          PropertyKeyFormat.WORKER_TIERED_STORE_LEVEL_RESERVED_RATIO_FORMAT.format(ordinal);
       String tierAlias = mStorageTierAssoc.getAlias(ordinal);
-      long reservedSpaceBytes =
-          (long) (capOnTiers.get(tierAlias) * Configuration.getDouble(tierReservedSpaceProp));
-      mBytesToReserveOnTiers.put(tierAlias, reservedSpaceBytes + lastTierReservedBytes);
-      lastTierReservedBytes += reservedSpaceBytes;
+      // HighWatemark defines when to start the space reserving process
+      PropertyKey tierHighWatermarkProp =
+          PropertyKeyFormat.WORKER_TIERED_STORE_LEVEL_HIGH_WATERMARK_RATIO_FORMAT.format(ordinal);
+      long highWatermarkInBytes =
+          (long) (capOnTiers.get(tierAlias) * Configuration.getDouble(tierHighWatermarkProp));
+
+      // LowWatemark defines when to stop the space reserving process if started
+      PropertyKey tierLowWatermarkProp =
+          PropertyKeyFormat.WORKER_TIERED_STORE_LEVEL_LOW_WATERMARK_RATIO_FORMAT.format(ordinal);
+      long capOnTier = capOnTiers.get(tierAlias);
+      long lowWatermarkInBytes =
+          (long) (capOnTier - capOnTier * Configuration.getDouble(tierLowWatermarkProp));
+      mHighWaterMarkInBytesOnTiers.put(tierAlias, highWatermarkInBytes);
+      mLowWaterMarkInBytesOnTiers.put(tierAlias, lowWatermarkInBytes + lastTierReservedBytes);
```

Comment:
@maobaolong When the high/low watermarks are populated, could you add a check to ensure high > low? If not we should at least warn the user of the misconfiguration.